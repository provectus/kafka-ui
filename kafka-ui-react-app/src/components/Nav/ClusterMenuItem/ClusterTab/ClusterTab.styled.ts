import styled from 'styled-components';

export const StyledClusterTab = styled.li`
  font-size: 14px;
  font-weight: 500;
  user-select: none;

  & .cluster-tab-wrapper {
    padding: 0.5em 0.75em;
    cursor: pointer;
    text-decoration: none;
    margin: 0px 0px;
    line-height: 20px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    gap: 10px;
    background-color: ${(props) =>
      props.theme.menuStyles.backgroundColor.normal};
    color: ${(props) => props.theme.menuStyles.color.normal};

    &:hover {
      background-color: ${(props) =>
        props.theme.menuStyles.backgroundColor.hover};
      color: ${(props) => props.theme.menuStyles.color.hover};
    }

    & .cluster-tab-l {
      display: flex;
      align-items: center;
      gap: 10px;
    }
  }
`;
