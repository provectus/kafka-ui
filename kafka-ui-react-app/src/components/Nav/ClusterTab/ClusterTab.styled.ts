import styled, { css } from 'styled-components';
import { ServerStatus } from 'generated-sources';

export const Wrapper = styled.li.attrs({ role: 'menuitem' })<{
  isOpen: boolean;
}>(
  ({ theme, isOpen }) => css`
    font-size: 14px;
    font-weight: 500;
    user-select: none;

    display: grid;
    grid-template-columns: min-content min-content auto min-content;
    grid-template-areas: 'title status . chevron';
    gap: 0 5px;

    padding: 0.5em 0.75em;
    cursor: pointer;
    text-decoration: none;
    margin: 0;
    line-height: 20px;
    align-items: center;
    color: ${isOpen ? theme.menu.color.isOpen : theme.menu.color.normal};
    background-color: ${theme.menu.backgroundColor.normal};

    &:hover {
      background-color: ${theme.menu.backgroundColor.hover};
      color: ${theme.menu.color.hover};
    }
  `
);

export const Title = styled.div`
  grid-area: title;
  white-space: nowrap;
  max-width: 110px;
  overflow: hidden;
  text-overflow: ellipsis;
  color: ${({ theme }) => theme.menu.titleColor};
`;

export const StatusIconWrapper = styled.svg.attrs({
  viewBox: '0 0 4 4',
  xmlns: 'http://www.w3.org/2000/svg',
})`
  grid-area: status;
  fill: none;
  width: 4px;
  height: 4px;
`;

export const StatusIcon = styled.circle.attrs({
  cx: 2,
  cy: 2,
  r: 2,
  role: 'status-circle',
})<{ status: ServerStatus }>(({ theme, status }) => {
  const statusColor: {
    [k in ServerStatus]: string;
  } = {
    [ServerStatus.ONLINE]: theme.menu.statusIconColor.online,
    [ServerStatus.OFFLINE]: theme.menu.statusIconColor.offline,
    [ServerStatus.INITIALIZING]: theme.menu.statusIconColor.initializing,
  };

  return css`
    fill: ${statusColor[status]};
  `;
});

export const ChevronWrapper = styled.svg.attrs({
  viewBox: '0 0 10 6',
  xmlns: 'http://www.w3.org/2000/svg',
})`
  grid-area: chevron;
  width: 10px;
  height: 6px;
  fill: none;
`;

type ChevronIconProps = { $open: boolean };

export const ChevronIcon = styled.path.attrs<ChevronIconProps>(({ $open }) => ({
  d: $open ? 'M8.99988 5L4.99988 1L0.999878 5' : 'M1 1L5 5L9 1',
}))<ChevronIconProps>`
  stroke: ${({ theme }) => theme.menu.chevronIconColor};
`;
