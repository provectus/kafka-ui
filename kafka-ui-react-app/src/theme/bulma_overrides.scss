.has {
  &-text-overflow-ellipsis {
    flex: 1;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  &-margin {
    &-right {
      margin-right: 10px;
    }
  }

  &-content-overflow-ellipsis {
    max-height: 73px;
    overflow: hidden;
    text-overflow: ellipsis;
    background:
      -webkit-linear-gradient(90deg, rgba(0, 0, 0, .1) 0%, rgba(0, 0, 0, 1) 40%);
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }
}

.breadcrumb li {
  &.is-active > span {
    padding: 0 0.75em;
  }

  &:first-child > span {
    padding-left: 0;
  }
}

.section {
  animation: fadein .5s;
}

.select.is-block select {
  width: 100%;
}

.notification {
  &.is-light {
    &.is-primary {
      background-color: #ebfffc;
      color: #00947e;
    }

    &.is-danger {
      background-color: #feecf0;
      color: #cc0f35;
    }
  }
}

@keyframes fadein {
  from { opacity: 0; }
  to   { opacity: 1; }
}

.level.level-multiline {
  flex-wrap: wrap;
  .level-item.is-one-third {
    flex-basis: 26%;
  }
  .level-item.is-one-third:nth-child(n+4) {
    margin-top: 10px;
  }
}
