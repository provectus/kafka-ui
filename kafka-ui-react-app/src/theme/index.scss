@import 'src/theme/bulma_overrides';

#root, body, html {
  width: 100%;
  position: relative;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background: repeating-linear-gradient(
    145deg,
    rgba(0,0,0,.003),
    rgba(0,0,0,.005) 5px,
    rgba(0,0,0,0) 5px,
    rgba(0,0,0,0) 10px
  ),
  repeating-linear-gradient(
    -145deg,
    rgba(0,0,0,.003),
    rgba(0,0,0,.005) 5px,
    rgba(0,0,0,0) 5px,
    rgba(0,0,0,0) 10px
  );
  background-color: $light;

}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

.cursor-pointer {
  cursor: pointer;
}
