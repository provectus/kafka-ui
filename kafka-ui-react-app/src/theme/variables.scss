// Typography
$size-7: 0.875rem;
$size-8: 0.75rem;
$body-line-height: 1.34;
$body-color: hsl(0, 100%, 0%);

// Section
$section-padding-desktop: 0 0 0.5rem;
$section-padding: 1rem 1.5rem 3rem;

// Tag
$tag-radius: 16px;
