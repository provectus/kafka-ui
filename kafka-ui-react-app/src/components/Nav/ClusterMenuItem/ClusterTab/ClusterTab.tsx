import { StyledClusterTab } from 'components/Nav/ClusterMenuItem/ClusterTab/ClusterTab.styled';
import { ServerStatus } from 'generated-sources';
import React from 'react';
import ClusterStatusIcon from 'components/Nav/ClusterStatusIcon';
import DefaultClusterIcon from 'components/Nav/DefaultClusterIcon';

import ClusterTabChevron from './ClusterTabChevron';

export interface ClusterTabProps {
  title?: string;
  status: ServerStatus;
  defaultCluster?: boolean;
  isOpen: boolean;
  toggleClusterMenu: () => void;
}

const ClusterTab: React.FC<ClusterTabProps> = ({
  status,
  title,
  defaultCluster,
  isOpen,
  toggleClusterMenu,
}) => {
  return (
    <StyledClusterTab onClick={() => toggleClusterMenu()}>
      <div className="cluster-tab-wrapper">
        <div className="cluster-tab-l">
          {defaultCluster && <DefaultClusterIcon />}
          {title}
          <ClusterStatusIcon status={status} />
        </div>
        <ClusterTabChevron isOpen={isOpen} />
      </div>
    </StyledClusterTab>
  );
};

export default ClusterTab;
