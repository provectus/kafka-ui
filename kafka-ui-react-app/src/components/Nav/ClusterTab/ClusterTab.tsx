import React from 'react';
import { ServerStatus } from 'generated-sources';

import * as S from './ClusterTab.styled';

export interface ClusterTabProps {
  title?: string;
  status: ServerStatus;
  isOpen: boolean;
  toggleClusterMenu: () => void;
}

const ClusterTab: React.FC<ClusterTabProps> = ({
  status,
  title,
  isOpen,
  toggleClusterMenu,
}) => (
  <S.Wrapper onClick={toggleClusterMenu} isOpen>
    <S.Title title={title}>{title}</S.Title>

    <S.StatusIconWrapper>
      <S.StatusIcon status={status} aria-label="status">
        <title>{status}</title>
      </S.StatusIcon>
    </S.StatusIconWrapper>

    <S.ChevronWrapper>
      <S.ChevronIcon $open={isOpen} />
    </S.ChevronWrapper>
  </S.Wrapper>
);

export default ClusterTab;
