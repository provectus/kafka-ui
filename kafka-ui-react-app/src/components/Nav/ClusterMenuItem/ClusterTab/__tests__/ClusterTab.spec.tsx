import { render } from '@testing-library/react';
import ClusterTab, {
  ClusterTabProps,
} from 'components/Nav/ClusterMenuItem/ClusterTab/ClusterTab';
import { ServerStatus } from 'generated-sources';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from 'theme/theme';

describe('ClusterTab component', () => {
  const setupWrapper = (props?: Partial<ClusterTabProps>) => (
    <ThemeProvider theme={theme}>
      <ClusterTab
        status={ServerStatus.ONLINE}
        isOpen
        toggleClusterMenu={jest.fn()}
        {...props}
      />
    </ThemeProvider>
  );
  describe('when online', () => {
    it('matches the snapshot', () => {
      const component = render(setupWrapper());
      expect(component.baseElement).toMatchSnapshot();
    });
  });

  describe('when offline', () => {
    it('matches the snapshot', () => {
      const component = render(
        setupWrapper({ status: ServerStatus.OFFLINE, isOpen: false })
      );
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
