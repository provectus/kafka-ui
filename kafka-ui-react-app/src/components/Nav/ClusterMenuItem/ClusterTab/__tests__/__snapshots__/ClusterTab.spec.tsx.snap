// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterTab component when offline matches the snapshot 1`] = `
<body>
  <div>
    <li
      class="sc-bdfBQB ibdTPX"
    >
      <div
        class="cluster-tab-wrapper"
      >
        <div
          class="cluster-tab-l"
        >
          <svg
            fill="none"
            height="4"
            viewBox="0 0 4 4"
            width="4"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="2"
              cy="2"
              fill="#E51A1A"
              r="2"
            />
          </svg>
        </div>
        <svg
          class="cluster-tab-chevron"
          fill="none"
          height="6"
          viewBox="0 0 10 6"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 1L5 5L9 1"
            stroke="#73848C"
          />
        </svg>
      </div>
    </li>
  </div>
</body>
`;

exports[`ClusterTab component when online matches the snapshot 1`] = `
<body>
  <div>
    <li
      class="sc-bdfBQB ibdTPX"
    >
      <div
        class="cluster-tab-wrapper"
      >
        <div
          class="cluster-tab-l"
        >
          <svg
            fill="none"
            height="4"
            viewBox="0 0 4 4"
            width="4"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="2"
              cy="2"
              fill="#5CD685"
              r="2"
            />
          </svg>
        </div>
        <svg
          class="cluster-tab-chevron"
          fill="none"
          height="6"
          viewBox="0 0 10 6"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.99988 5L4.99988 1L0.999878 5"
            stroke="#73848C"
          />
        </svg>
      </div>
    </li>
  </div>
</body>
`;
